include/zephyr/bluetooth/ead.h

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+
+/** Additional Authenticated Data size in bytes */
+#define BT_EAD_AAD_SIZE	       1
+/** Randomizer size in bytes */
+#define BT_EAD_RANDOMIZER_SIZE 5
+/** nonce size in bytes */
+#define BT_EAD_NONCE_SIZE      13
+/** Key size in bytes */
+#define BT_EAD_KEY_SIZE	       16
+/** Initialisation Vector size in bytes */
+#define BT_EAD_IV_SIZE	       8
+/** MIC size in bytes */
+#define BT_EAD_MIC_SIZE	       4
+
+/**
+ * @brief Encrypt and authenticate the given advertising data. The resulting
+ * data in @p encrypted_ad_data will look like that:
+ *
+ *     - Randomizer is added in the 5 first bytes;
+ *     - AAD is added in the next byte;
+ *     - Encrypted payload is added (payload_size bytes);
+ *     - MIC is added in the last 4 bytes.
+ *
+ * @note The function must be called each time the RPA is updated or the data
+ * are modified.
+ *
+ * @note The term `advertising structure` is used to describe the advertising
+ *       data with the advertising type and the length of those two.
+ *
+ * @param[in]  session_key Key of 16 bytes used for the encryption.
+ * @param[in]  iv Initialisation Vector used to generate the nonce. It must be
+ *             changed each time the Session Key change.
+ * @param[in]  payload Advertising Data to encrypt. Can be multiple advertising
+ *             structure concatenated.
+ * @param[in]  payload_size Size of the Advertising Data to encrypt.
+ * @param[out] encrypted_ad_data Encrypted Adv Data including the Randomizer and
+ *             the MIC. Size must be `BT_EAD_RANDOMIZER_SIZE + BT_EAD_AAD_SIZE +
+ *             payload_size + BT_EAD_MIC_SIZE`.
```

Comment:
@theob-pro this comment is still valid

Reply:
Sorry, I forgot to reply. I don't think that is necessary now. Also the "header" is only the Randomizer, the MIC is at the end. Anyway, now that there is the helpers functions it's no longer needed. Do you agree?

Reply:
It was never "needed" :D But IMO it's easier to use a single value, rather than having to use two values to denote the additional size of the buffer. 

Alternatively we could also add a macro similar to `BT_ISO_SDU_BUF_SIZE` or `BT_L2CAP_BUF_SIZE` which are used for this exact same purpose.

Reply:
> But IMO it's easier to use a single value, rather than having to use two values to denote the additional size of the buffer.

I agree but with the `bt_ead_get_encrypted_data_size` and `bt_ead_get_decrypted_data_size` the user don't need to do calculation by itself. That said, I should maybe move out of the public header the variables that the user don't need anymore (`BT_EAD_AAD_SIZE`, `BT_EAD_RANDOMIZER_SIZE`, `BT_EAD_NONCE_SIZE`, `BT_EAD_MIC_SIZE`).

Reply:
Okay, I understand, I will add a macro to compute the EAD size based on the payload size.